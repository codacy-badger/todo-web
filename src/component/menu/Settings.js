import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCog } from "@fortawesome/free-solid-svg-icons";
import { faTimes } from "@fortawesome/free-solid-svg-icons";
import "component/menu/Settings.sass";

class Settings extends React.Component {
  state = {
    expanded: false
  };

  switchExpanded() {
    const expanded = this.state.expanded;
    this.setState({
      expanded: !expanded
    });
  }

  render() {
    const settingsIcon = this.state.expanded ? faTimes : faCog;
    return (
      <div
        className={`settings-component ${
          this.state.expanded ? "expanded" : ""
        }`}
      >
        <FontAwesomeIcon
          className="settings-component-button"
          icon={settingsIcon}
          onClick={() => this.switchExpanded()}
        />
        <div className="settings-component-panel">{this.props.children}</div>
      </div>
    );
  }
}

export default Settings;
