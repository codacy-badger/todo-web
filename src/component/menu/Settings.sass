@import "colors.sass"

.settings-component
  position: fixed
  top: 0
  right: 0
  transition: 0.5s ease all

  &.expanded
    .settings-component-button
      right: 60px

    .settings-component-panel
      width: 60px

  .settings-component-button
    position: absolute
    top: 0
    right: 10px
    display: inline-block
    margin: 20px
    width: 35px
    height: 35px
    line-height: 35px
    text-align: center
    border-radius: 100%
    color: $text-primary
    padding: 10px
    cursor: pointer
    font-size: 30px
    transition: 0.5s ease all

    &:hover
      background-color: $background-secondary
      box-shadow: 5px 5px 10px $shadow

  .settings-component-panel
    display: inline-block
    width: 0
    height: 100vh
    background-color: $background-secondary
    transition: 0.5s ease all
    text-align: center
    box-shadow: 0 0 10px $shadow
