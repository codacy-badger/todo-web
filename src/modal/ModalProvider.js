import React from "react";
import ConfirmationModal from "component/modal/ConfirmationModal";
import SelectModal from "component/modal/SelectModal";
import InputModal from "component/modal/InputModal";
import { CONFIRMATION, SELECT, INPUT } from "modal/types";
import { validateModalData } from "modal/validator";

export const ModalContext = React.createContext();

class ModalProvider extends React.Component {
  state = {
    type: CONFIRMATION,
    visible: false
  };

  openModal = modal => {
    const modalData = validateModalData(modal);
    this.setState(modalData);
  };

  closeModal = () => {
    this.setState({
      visible: false
    });
  };

  mountModalComponent() {
    switch (this.state.type) {
      case SELECT:
        return <SelectModal />;
      case INPUT:
        return <InputModal />;
      default:
        return <ConfirmationModal />;
    }
  }

  render() {
    return (
      <ModalContext.Provider
        value={{
          modal: this.state,
          openModal: this.openModal,
          closeModal: this.closeModal
        }}
      >
        {this.mountModalComponent()}
        {this.props.children}
      </ModalContext.Provider>
    );
  }
}

export default ModalProvider;
