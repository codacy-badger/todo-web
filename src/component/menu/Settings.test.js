import React from "react";
import Settings from "component/menu/Settings";
import renderer from "react-test-renderer";
import ModalProvider from "modal/ModalProvider";

describe("Settings component", () => {
  test("should render Settings", () => {
    const component = renderer
      .create(
        <ModalProvider>
          <Settings />
        </ModalProvider>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
