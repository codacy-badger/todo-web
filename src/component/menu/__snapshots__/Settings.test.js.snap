// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component should render Settings 1`] = `
<div
  className="settings-component "
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-cog fa-w-16 settings-component-button"
    data-icon="cog"
    data-prefix="fas"
    focusable="false"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <div
    className="settings-component-sidebar"
  >
    <div
      className="common-actions"
    >
      <div
        className="settings-item-component"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-home fa-w-18 "
          data-icon="home"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="settings-item-component"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user fa-w-14 "
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="settings-item-component"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-flag-usa fa-w-16 "
          data-icon="flag-usa"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 0C14.3 0 0 14.3 0 32v464c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V32C64 14.3 49.7 0 32 0zm267.9 303.6c-57.2-15.1-111.7-28.8-203.9 11.1V384c185.7-92.2 221.7 53.3 397.5-23.1 11.4-5 18.5-16.5 18.5-28.8v-36c-43.6 17.3-80.2 24.1-112.1 24.1-37.4-.1-68.9-8.4-100-16.6zm0-96c-57.2-15.1-111.7-28.8-203.9 11.1v61.5c94.8-37.6 154.6-22.7 212.1-7.6 57.2 15.1 111.7 28.8 203.9-11.1V200c-43.6 17.3-80.2 24.1-112.1 24.1-37.4 0-68.9-8.3-100-16.5zm9.5-125.9c51.8 15.6 97.4 29 202.6-20.1V30.8c0-25.1-26.8-38.1-49.4-26.6C291.3 91.5 305.4-62.2 96 32.4v151.9c94.8-37.5 154.6-22.7 212.1-7.6 57.2 15 111.7 28.7 203.9-11.1V96.7c-53.6 23.5-93.3 31.4-126.1 31.4s-59-7.8-85.7-15.9c-4-1.2-8.1-2.4-12.1-3.5V75.5c7.2 2 14.3 4.1 21.3 6.2zM160 128.1c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16-7.2 16-16 16zm0-55.8c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16zm64 47.9c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16zm0-55.9c-8.8 0-16-7.1-16-16 0-8.8 7.2-16 16-16s16 7.1 16 16c0 8.8-7.2 16-16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="settings-item-component"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-info fa-w-6 "
          data-icon="info"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
    <div
      className="component-actions"
    />
  </div>
</div>
`;
